### PR TITLE
Centre text in case where the distance between baselines is larger th…

### PR DESCRIPTION
**Pull request by jacobp100**
…an the font size (per CSS spec)

Extending upon work https://github.com/facebook/react-native/pull/7603
Closes https://github.com/facebook/react-native/pull/8851

screen shot 2016-08-04 at 14 49 21
screen shot 2016-08-04 at 14 49 24

### CODE REVIEW

**Comment by facebook-github-bot-8** [comment]
By analyzing the blame information on this pull request, we identified @rigdern and @emilsjolander to be potential reviewers.

File: Libraries/Text/RCTShadowText.m [hunk 1]
```diff
@@ -415,9 +418,16 @@ - (void)_setParagraphStyleOnAttributedString:(NSMutableAttributedString *)attrib
...
     paragraphStyle.minimumLineHeight = lineHeight;
     paragraphStyle.maximumLineHeight = lineHeight;
+
     [attributedString addAttribute:NSParagraphStyleAttributeName
                              value:paragraphStyle
                              range:(NSRange){0, attributedString.length}];
+
+    if (lineHeight > fontSize) {
+      [attributedString addAttribute:NSBaselineOffsetAttributeName
+                               value:@(lineHeight/2 - fontSize/2)
```
Comment (ide) [inline Libraries/Text/RCTShadowText.m]: Style: Spaces around operators (`/`)

File: Libraries/Text/RCTShadowText.m [hunk 2]
```diff
@@ -326,7 +326,9 @@ - (NSAttributedString *)_attributedStringWithFontFamily:(NSString *)fontFamily
   [self _addAttribute:NSFontAttributeName withValue:font toAttributedString:attributedString];
   [self _addAttribute:NSKernAttributeName withValue:letterSpacing toAttributedString:attributedString];
   [self _addAttribute:RCTReactTagAttributeName withValue:self.reactTag toAttributedString:attributedString];
-  [self _setParagraphStyleOnAttributedString:attributedString heightOfTallestSubview:heightOfTallestSubview];
+  [self _setParagraphStyleOnAttributedString:attributedString
+                                    fontSize:[font pointSize]
```
Comment (ide) [inline Libraries/Text/RCTShadowText.m]: Style: Use property notation (`.`) when the iOS docs document members like pointSize as properties.

fontsize

**Comment by ide** [comment]
I'm not super familiar with typography and am unsure if `(lineHeight - fontSize) / 2` is the right calculation for the offset. In CSS at least, fontSize apparently is not directly related to the height and MDN says: The font-size CSS property specifies the size of the font (historically the width of the capital "M"). I believe we want a calculation closer to `(lineHeightWeAreGoingToUse - lineHeightIntrinsicToThisText) / 2`.

**Comment by facebook-github-bot-6** [comment]
@jacobp100 updated the pull request.

**Comment by jacobp100** [comment]
You’re correct in that. I’ve updated it to use the font `lineHeight` property.

**Comment by ide** [comment]
Looks like the Obj-C tests are failing (UIExplorerSnapshotTests in https://travis-ci.org/facebook/react-native/jobs/149979798) -- some of the tests capture screenshots and do a before-and-after comparison, so you probably need to re-record them.

See https://facebook.github.io/react-native/releases/0.31/docs/testing.html#snapshot-tests-ios for more about the screenshot tests.

**Comment by facebook-github-bot-3** [comment]
@jacobp100 updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@jacobp100 updated the pull request.

**Comment by facebook-github-bot-7** [comment]
@jacobp100 updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@jacobp100 updated the pull request.

**Comment by ide** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-4** [comment]
Thanks for importing.If you are an FB employee go to [Phabricator](https://phabricator.intern.facebook.com/D3706347) to review internal test results.

**Pull request closed without merging**